### PR TITLE
lvm2: 2.03.23 -> 2.03.25

### PR DESCRIPTION
**Pull request by ajs124**
## Description of changes
https://gitlab.com/lvmteam/lvm2/-/blob/v2_03_24/WHATS_NEW

## Things done
- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).
---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by 7c6f434c** [comment]
@ofborg test lvm2

**Comment by ajs124** [comment]
Draft, because it breaks something... pkgsStatic? no, I think that one worked. maybe pkgsMusl?

should have put it in the PR description.

**Comment by 7c6f434c** [comment]
kvdo build seems to fail

File: pkgs/os-specific/linux/lvm2/common.nix [hunk 1]
```diff
@@ -101,10 +101,6 @@ stdenv.mkDerivation rec {
       multipath_tools = optionalTool enableMultipath multipath-tools;
       vdo = optionalTool enableVDO vdo;
     }))
-    # Musl fix from Alpine
-    ./fix-stdio-usage.patch
```
Comment (alyssais) [inline pkgs/os-specific/linux/lvm2/common.nix:105]: This is still needed.

File: nixos/modules/tasks/lvm.nix [hunk 2]
```diff
@@ -89,7 +89,7 @@ in {
     (mkIf cfg.boot.vdo.enable {
       boot = {
         initrd = {
-          kernelModules = [ "kvdo" ];
+          kernelModules = [ "dm-vdo" ];
```
Comment (alyssais) [inline nixos/modules/tasks/lvm.nix:92]: Shouldn't we have an assertion that if this is enabled the kernel is a recent enough version?

**Review by 7c6f434c — APPROVED** [review]

**Comment by 7c6f434c** [comment]
OfBorg failure: unrelated coreutils failure because symlink loop got a change in the error reporting breaking a test.

**Comment by 7c6f434c** [comment]
In any case the failure seems quite far from these changes…

**Comment by Atemu** [comment]
It appears this broke boot for at least one user: https://github.com/NixOS/nixpkgs/issues/346935

I think we should revert until that situation is cleared up.

**Pull request merged**